SU-HSDO/suhumsci
# STN-325: Private Pages

# READY FOR REVIEW

## Summary
This pull request adds the following:
- A lock icon to the left of the title in private pages
- A lock icon to the left of the link text for private pages in the main nav and menu blocks
- A utility class (`hs-private-link`) that adds a lock icon to the left of link text in WYSIWYG editors

Work has been reviewed in the following browsers:

This has been given a design approval from J. All icons should now be black instead of the primary color so they don't exactly match the screenshots below.

## Need Review By (Date)
6/25

## Urgency
medium

## Steps to Test
1. Run `npm test`
2. Import changes to the ckeditor config files
3. Create a new private page (content -> private page) and add the page to the main navigation
4. - [x] There should be a lock icon next the link the main and secondary navigation regardless of the nesting level (see video in Jira card for more details on what this should look like)
    <img width="1298" alt="Private_Page___Department_of_Economics_and_Slack_____-general___Sparkbox" src="https://user-images.githubusercontent.com/2486846/85432907-a006cc80-b551-11ea-8e21-69f58ca73a03.png">
5. - [x] There should be a lock icon next to the title of the private page.
    <img width="1298" alt="Private_Page___Department_of_Economics_and_Slack_____-general___Sparkbox" src="https://user-images.githubusercontent.com/2486846/85432862-941b0a80-b551-11ea-8fc7-68f4c3aea916.png">
6. - [x] There should be an option to add the `hs-private-link` class to inline links in the ckeditor
    <img width="1309" alt="Screen_Shot_2020-06-23_at_1_45_04_PM" src="https://user-images.githubusercontent.com/2486846/85437287-5d94be00-b558-11ea-88f9-7dfcfcda4afa.png">
    <img width="1315" alt="Private_Page___Department_of_Economics" src="https://user-images.githubusercontent.com/2486846/85433407-66829100-b552-11ea-8949-83683c59c859.png">


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
